### PR TITLE
Close envs in async training

### LINKED ISSUE #342: How to close environments in experiments.train_agent_async
My environment needs to be closed after its use, but I couldn't find the way to close environments distributed in `experiments.train_agent_async` after training is finished.

### PR DESCRIPTION
**Pull request by muupan**
- Resolves #342 
- Corrects the termination condition of `train_agent_async`.
- Adds tests for `train_agent_async`

I confirmed that all the tests pass.

### CODE REVIEW

**Comment by muupan** [comment]
It seems that `eval_env` need to be closed, too.

**Comment by muupan** [comment]
I confirmed again that all the tests pass.

File: chainerrl/experiments/train_agent_async.py [hunk 1]
```diff
@@ -87,7 +87,7 @@ def train_loop(process_idx, env, agent, steps, outdir, counter,
                 for hook in global_step_hooks:
                     hook(env, agent, global_t)
 
-                if global_t > steps or training_done.value:
+                if global_t >= steps or training_done.value:
```
Comment (toslunar) [inline chainerrl/experiments/train_agent_async.py]: Could you also fix the line 101?

Comment (muupan) [reply chainerrl/experiments/train_agent_async.py]: Fixed and added a test of it.

**Pull request merged**